src/resmom/mom_main.c

```diff
@@ -6732,11 +6733,9 @@ calc_cpupercent(job *pjob, u_long oldcput, u_long newcput, time_t sampletime)
 		rd = find_resc_def(svr_resc_def, "Walltime", svr_resc_size);
 		assert(rd != NULL);
 		preswalltime = find_resc_entry(at_used, rd);
-		assert(rd != NULL);
-		if (preswalltime != NULL) {
+		if (preswalltime != NULL &&
+			(preswalltime->rs_value.at_flags & ATR_VFLAG_SET) != 0) {
```

Comment:
I prefer that you parenthesize explicitly the 2 conditions inside if (i.e. preswalltime != NULL , preswalltime->rs_value.at_flags & ...) so that the operation precedence is obvious.